contracts/oracle/src/contract/feed.rs

```diff
@@ -77,20 +83,88 @@ where
         Ok(prices)
     }
 
-    fn calc_all_prices(
-        &self,
-        storage: &dyn Storage,
+    pub fn all_prices_iter<'a>(
+        self,
+        storage: &'a dyn Storage,
+        tree: &'a SupportedPairs<OracleBase>,
         at: Timestamp,
         total_feeders: usize,
-    ) -> Result<Vec<SpotPrice>, ContractError> {
-        let tree: SupportedPairs<OracleBase> = SupportedPairs::load(storage)?;
-        let mut prices = vec![];
-        for leg in tree.clone().query_supported_pairs() {
-            if let Ok(price) = self.calc_price(&tree, storage, &leg.from, at, total_feeders) {
-                prices.push(price);
+    ) -> Result<impl Iterator<Item = BasePrice<OracleBase, SwapGroup>> + 'a, ContractError> {
+        struct LegCmd<'a, 'b, OracleBase>
+        where
+            OracleBase: Currency,
+        {
+            feeds: &'a PriceFeeds<'static>,
+            storage: &'a dyn Storage,
+            at: Timestamp,
+            total_feeders: usize,
+            stack: &'b mut Vec<BasePrice<OracleBase, SwapGroup>>,
```

Comment:
Not sure I understand the need of storing them all. Besides the extra memory that requires probably a not-so-cheap conversion back into Price<C1, C2>.

Reply:
clarified offline, we do need it. Just agreed to hide it as part of the state.